spec/requests/provider_management/external_users_spec.rb

```diff
@@ -128,33 +160,57 @@
       it { expect { enable_user }.to change { external_user.reload.enabled? }.from(false).to(true) }
       it { expect { enable_user }.to change { external_user.reload.disabled_at }.from(be_kind_of(Time)).to(nil) }
 
-      # TODO: check - is provider memdership matching relevant? how would a unique external user ever be in an unmatching provider?
-      context 'when external user belongs to a different provider' do
+      context 'when successfull response' do
+        before { enable_user }
+
+        it { expect(external_user.reload).to be_enabled }
+        it { expect(response).to redirect_to provider_management_provider_external_user_path }
+        it { expect(flash[:notice]).to eq('User successfully enabled') }
+      end
+
+      # TODO: is this behaviour needed? when would a external user ever be from another provider at this point
```

Comment:
This isn't so much behaviour that is needed but rather guarding against badly formed urls. The update is called with a PATCH request to;

```
/provider_management/providers/:provider_id/external_users/:id/update_availability
```

where `:id` is the id of an external user belonging to the provider `:provider_id`. The test checks that with a badly formed url, where the `:id` does not correspond to `:provider_id`, then the behaviour is predictable and, specifically, no update is made. If the behaviour is not predictable then there is a possibility of a security exploit.

This is related to [CFP-484](https://dsdmoj.atlassian.net/browse/CFP-484) which is regarding viewing the external user page.